tests/test_photon_ops.py

```diff
@@ -328,28 +430,74 @@ def test_config_lsst_diffraction():
     galsim.config.BuildPhotonOps(config["stamp"], "photon_ops", config)
 
 
-def test_config_lsst_optics():
-    """Check the config interface to BatoidPhotonOps."""
+def test_config_lsst_diffraction_optics():
+    """Check the config interface to LsstDiffractionOptics."""
 
-    boresight = galsim.CelestialCoord(
-        1.1047934165124105 * galsim.radians, -0.5261230452954583 * galsim.radians
-    )
     config = {
-        "input": {
-            "telescope": {
-                "file_name":"LSST_r.yaml",
-            }
+        **TEST_BASE_CONFIG,
+        **TEST_OPSIM_META_CONFIG,
+        "det_name": "R22_S11",
+        "image_pos": galsim.PositionD(
+            3076.4462608524213, 1566.4896702703757
+        ),  # This would get set appropriately during normal config processing.
+        "stamp": {
+            "photon_ops": [
+                {
+                    "type": "lsst_diffraction_optics",
+                    "camera": "LsstCam",
+                    "boresight": {
+                        "type": "RADec",
+                        "ra": "1.1047934165124105 radians",
+                        "dec": "-0.5261230452954583 radians",
+                    },
+                    "latitude": -30.24463,
+                }
+            ]
         },
-        "sky_pos": {
-            "type": "RADec",
-            "ra": "1.1056660811384078 radians",
-            "dec": "-0.5253441048502933 radians",
+    }
+    galsim.config.ProcessInput(config)
+    galsim.config.BuildPhotonOps(config["stamp"], "photon_ops", config)
+
+
+def test_config_lsst_diffraction_optics_without_field_rotation():
+    """Check the config interface to LsstDiffractionOptics."""
+
+    config = {
+        **TEST_BASE_CONFIG,
+        **TEST_OPSIM_META_CONFIG,
+        "det_name": "R22_S11",
+        "image_pos": galsim.PositionD(
+            3076.4462608524213, 1566.4896702703757
+        ),  # This would get set appropriately during normal config processing.
+        "stamp": {
+            "photon_ops": [
+                {
+                    "type": "lsst_diffraction_optics",
+                    "camera": "LsstCam",
+                    "boresight": {
+                        "type": "RADec",
+                        "ra": "1.1047934165124105 radians",
+                        "dec": "-0.5261230452954583 radians",
+                    },
+                    "latitude": -30.24463,
+                    "disable_field_rotation": True,
+                }
+            ]
         },
+    }
+    galsim.config.ProcessInput(config)
+    galsim.config.BuildPhotonOps(config["stamp"], "photon_ops", config)
```

Comment:
These tests don't seem to actually test anything other than that no exceptions were raised.  Can we add some actual tests of the images here?

Reply:
I would expect there to be some test that the code you changed is doing what you think it should do.

Reply:
If it is only that, I tested the code I changed that it is doing what I think it should do by `test_rubin_diffraction_optics_is_same_as_diffraction_and_optics`, `test_rubin_diffraction_produces_spikes` and `test_rubin_diffraction_does_not_show_field_rotation_when_deactivated`.
The deserialization tests make sure that I obtain instances of `RubinDiffraction`, `RubinDiffractionOptics` and `RubinOptics`. From there on the 3 above tests take over.
Indeed, there is a small gap in coverage: I dont test that the values from config end up in the instance being what I expect.
On the other hand side, this is actually should already be covered by galsim.
Still, I could add checks for the values in the instances after deserializing or I also could repeat the 3 configless tests from above.

Reply:
There ought to be some test that the image produced here is correct.  I'll leave to you to decide what that means, but right now this test (and the other two like it) is incomplete.

Reply:
Added a commit which compares the behavior of the deserialized photon ops to some reference ops